pkg/manager/member/pump_member_manager.go

```diff
@@ -284,23 +284,29 @@ func getNewPumpHeadlessService(tc *v1alpha1.TidbCluster) *corev1.Service {
 
 // getNewPumpConfigMap returns a configMap for pump
 func getNewPumpConfigMap(tc *v1alpha1.TidbCluster) (*corev1.ConfigMap, error) {
-	if tc.Spec.Pump == nil {
```

Comment:
Why remove this?

Reply:
I think the field should be verified before calling the function, just like other `getXXXConfigMap`